File: AC/Others/HHKB20201010/c.go
package main

import "fmt"

const MAX = 200001

func main() {
	var mmin, n int
	fmt.Scan(&n)
	p := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&p[i])
	}
	nums := make([]bool, MAX)
	for i := 0; i < n; i++ {
		nums[p[i]] = true
		if nums[mmin] == true {
			mmin = min(nums, mmin+1, p[i])
		}
		fmt.Println(mmin)
	}
}

func min(nums []bool, start, hit int) int {
	for i := start; i < len(nums); i++ {
		if nums[i] == false {
			return i
		}
	}
	return 0
}

File: AC/Others/HHKB20201010/c.py
from typing import List


def min(arr: List[int], hit: int) -> int:
    size = len(arr)
    for j in range(0, size):
        if arr[j] >= 0:
            return j


n = int(input())
p = list(map(int, input().split()))
MAX = 2000001
arr = [1]*MAX
mmin = 0
for i in range(mmin, n):
    arr[p[i]] = -1
    if arr[mmin] < 0:
        mmin = min(arr, p[i])
    print(mmin)
